builder/lxc/communicator.go

```diff
@@ -110,8 +111,13 @@ func (c *LxcAttachCommunicator) DownloadDir(src string, dst string, exclude []st
 
 func (c *LxcAttachCommunicator) Execute(commandString string) (*exec.Cmd, error) {
 	log.Printf("Executing with lxc-attach in container: %s %s %s", c.ContainerName, c.RootFs, commandString)
+
+	attachCommand := []string{"sudo", "lxc-attach"}
+	attachCommand = append(attachCommand, c.AttachOptions...)
+	attachCommand = append(attachCommand, []string{"--name", "%s", "--", "/bin/sh -c \"%s\""}...)
+
 	command, err := c.CmdWrapper(
-		fmt.Sprintf("sudo lxc-attach --name %s -- /bin/sh -c \"%s\"", c.ContainerName, commandString))
+		fmt.Sprintf(strings.Join(attachCommand, " "), c.ContainerName, commandString))
```

Comment:
I'd certainly be fine with using an array rather than a string here, or with using a proper shell-quoting method.  Apart from security issues, joining with `" "` won't work properly if the user-supplied `attach_options` have parameters with spaces in their values, which is a potential source of confusion and frustration.  

However, wouldn't this necessitate changing the `CmdWrapper` type to accept a slice rather than a string?  Also, there are a number of other locations in `communicator.go` subject to the same problem (e.g., the variations on `fmt.Sprintf("sudo cp %s %s", ...)`, which I mention not to excuse introducing a new instance of the issue, but rather to suggest that the issue might be best addressed with a dedicated PR (which I would be happy to submit :).